like the other CHOST patches, but this one catches another case where
$host is used (instead of the cannonicalised variables)

--- aclocal.m4
+++ aclocal.m4
@@ -93,10 +93,10 @@
         exeext='.exe'
         soext='.dll'
         ;;
-    i386-apple-darwin|powerpc-apple-darwin)
+    i386-apple-darwin*|powerpc-apple-darwin*)
         soext='.dylib'
         ;;
-    x86_64-apple-darwin)
+    x86_64-apple-darwin*)
         soext='.dylib'
         ;;
     esac
